### DISCUSSION TITLE
ERROR SMTPClient - SMTPSessionNotStartedError

### DISCUSSION THREAD

**Discussion by mascor** [postalserver/postal#2917] [Help with using Postal]
Hi,
sometimes the email is not sent and is labeled with a "SOFT FAIL".
The logs report this error:
_"ERROR SMTPClient::Endpoint::SMTPSessionNotStartedError: SMTPClient::Endpoint::SMTPSessionNotStartedError log_id=TKR6WEB9 component=worker thread=work0 original_queued_message=98991 queued_message=99008"_

What are these errors due to?
Thanks

**Comment by willpower232** [comment]
Where are you seeing that information? Can you share a screenshot?

**Reply by mascor** [reply]
<img width="943" alt="Screenshot 2024-04-09 alle 14 38 42" src="https://github.com/postalserver/postal/assets/9699878/eb04c715-4b01-43c8-b248-32f14a02b119">

**Reply by willpower232** [reply]
Interesting, you should be able to find that support ref in the logs for the worker so that would help clarify if it was a problem within Postal (i.e. you would see a whole stack trace) or a problem with google (unlikely but you never know).

The stack trace would also identify which part of Postal went wrong.

**Reply by mascor** [reply]
This is a part of the log that I can retrieve with the command "postal logs":

worker_1  | 2024-04-06 02:27:13 +0000 INFO   Sending message 1::98866 to <hidden_data>@gmail.com log_id=WSBSBYIF component=worker thread=work0 original_queued_message=98863 queued_message=98866
worker_1  | 2024-04-06 02:27:13 +0000 ERROR  SMTPClient::Endpoint::SMTPSessionNotStartedError: SMTPClient::Endpoint::SMTPSessionNotStartedError log_id=WSBSBYIF component=worker thread=work0 original_queued_message=98863 queued_message=98866
worker_1  | 2024-04-06 02:27:13 +0000 INFO   message requeued for trying later retry_after=2024-04-06 09:39:44 UTC component=worker thread=work0 original_queued_message=98863 queued_message=98866

**Reply by willpower232** [reply]
oh yeah is it also in the SMTP logs? if you look for `WSBSBYIF` in both, hopefully you'll see some extra context

**Reply by mascor** [reply]
Hi, if I try to execute:
`postal logs smtp`
it shows only this:
<img width="744" alt="Screenshot 2024-04-11 alle 08 02 46" src="https://github.com/postalserver/postal/assets/9699878/b459e104-de59-41d4-8f21-d95ea81fe8a9">

**Reply by willpower232** [reply]
Have a look for `WSBSBYIF` in the worker log and see if you can find all related rows, it should clarify what happened with that message

**Reply by jarqvi** [reply]
I had the same problem, but after a little research, I realized that the domain I want to send an email from has unconfirmed DNS records. Actually, the problem was with the DNS records of the domain I wanted to send an email from.

**Reply by jmduke** [reply]
I'm running into a similar issue. It's not impacting _all_ traffic, but a non-trivial portion. Here's what I see in logs:

```
worker-1  | 2024-09-05 11:45:06 +0000 ERROR  Net::SMTPServerBusy: 421 Max message per connection reached, closing transmission channel tnmpmscs
worker-1  |  log_id=YPXELCKH component=worker thread=work0 original_queued_message=441819 queued_message=442089
worker-1  | 2024-09-05 11:45:06 +0000 INFO   message requeued for trying later retry_after=2024-09-05 11:50:06 UTC component=worker thread=work0 original_queued_message=441819 queued_message=442089
worker-1  | 2024-09-05 11:45:06 +0000 INFO   parsing message content as it hasn't been parsed before component=worker thread=work0 original_queued_message=441819 queued_message=442096
worker-1  | 2024-09-05 11:45:06 +0000 INFO   Sending message 1::442096 to redacted1@yahoo.com log_id=YPXELCKH component=worker thread=work0 original_queued_message=441819 queued_message=442096
worker-1  | 2024-09-05 11:45:06 +0000 ERROR  SMTPClient::Endpoint::SMTPSessionNotStartedError: SMTPClient::Endpoint::SMTPSessionNotStartedError log_id=YPXELCKH component=worker thread=work0 original_queued_message=441819 queued_message=442096
worker-1  | 2024-09-05 11:45:06 +0000 INFO   message requeued for trying later retry_after=2024-09-05 11:50:06 UTC component=worker thread=work0 original_queued_message=441819 queued_message=442096
worker-1  | 2024-09-05 11:45:06 +0000 INFO   parsing message content as it hasn't been parsed before component=worker thread=work0 original_queued_message=441819 queued_message=442098
worker-1  | 2024-09-05 11:45:06 +0000 INFO   Sending message 1::442099 to redacted2@yahoo.com log_id=YPXELCKH component=worker thread=work0 original_queued_message=441819 queued_message=442098
worker-1  | 2024-09-05 11:45:06 +0000 ERROR  SMTPClient::Endpoint::SMTPSessionNotStartedError: SMTPClient::Endpoint::SMTPSessionNotStartedError log_id=YPXELCKH component=worker thread=work0 original_queued_message=441819 queued_message=442098
worker-1  | 2024-09-05 11:45:06 +0000 INFO   message requeued for trying later retry_after=2024-09-05 11:50:06 UTC component=worker thread=work0 original_queued_message=441819 queued_message=442098
worker-1  | 2024-09-05 11:45:06 +0000 INFO   parsing message content as it hasn't been parsed before component=worker thread=work0 original_queued_message=441819 queued_message=442105
worker-1  | 2024-09-05 11:45:06 +0000 INFO   Sending message 1::442105 to redacted3@yahoo.com log_id=YPXELCKH component=worker thread=work0 original_queued_message=441819 queued_message=442105
worker-1  | 2024-09-05 11:45:06 +0000 ERROR  SMTPClient::Endpoint::SMTPSessionNotStartedError: SMTPClient::Endpoint::SMTPSessionNotStartedError log_id=YPXELCKH component=worker thread=work0 original_queued_message=441819 queued_message=442105
worker-1  | 2024-09-05 11:45:06 +0000 INFO   message requeued for trying later retry_after=2024-09-05 11:50:06 UTC component=worker thread=work0 original_queued_message=441819 queued_message=442105
```

**Reply by jmduke** [reply]
I suspect this might actually be related to / solved by https://github.com/postalserver/postal/pull/3031?

**Comment by solracsf** [comment]
Same problem here.

![image](https://github.com/user-attachments/assets/329454f0-b604-4d4e-9f07-9d42be6f8a91)

**Comment by niels-przybilla** [comment]
Hello,

I have. similar issue ... Only with one domain I am sending too.

It started like this:

SMTPClient::Endpoint::SMTPSessionNotStartedError

Time: 0.0s

Support Ref: 3YVV9AWS

and then later it transforms to

Net::ReadTimeout with #<TCPSocket:(closed)>

Time: 60.3s

Support Ref: 3YVV9AWS

Any idea?

many thanks for your help.

**Reply by UEAB-BL** [reply]
I think Im having a similar issue, posted about it here:
https://github.com/orgs/postalserver/discussions/3232
Almost feels like something is done improperly and then we get IP blocked or something.

**Comment by plancked** [comment]
I'm also getting this, and it looks like it's only from Yahoo servers.
It's a temporary ban? why though?
